Sofa/framework/Core/src/sofa/core/topology/TopologyHandler.h

```diff
@@ -137,19 +137,16 @@ class SOFA_CORE_API TopologyHandler : public sofa::core::objectmodel::DDGNode
     virtual void ApplyTopologyChange(const TopologyChangeElementInfo<Topology::Tetrahedron>::EMoved* /*event*/) {}
     virtual void ApplyTopologyChange(const TopologyChangeElementInfo<Topology::Hexahedron>::EMoved* /*event*/) {}
 
-
-
-    virtual bool isTopologyDataRegistered() { return false; }
+    /// Method to notify that this topologyHandler is not anymore registerd into a Topology Container
+    void unRegisteredTopologyHandler() { m_isRegistered = false; }
+    /// Method to get the information if this topologyHandler is registered into a Topology Container
+    bool isTopologyHandlerRegistered() const { return m_isRegistered; }
```

Comment:
Yes and no.
Theoretically it could be possible to not use the bool and this method would return something like `m_topologyContainer->isTopologyHandlerRegistered(this)`  where isTopologyHandlerRegistered will do a find in its std::set of topologyHandler.

But I use this bool to "fake a smart pointer mechanism". 
This is needed because TopologyData/handler are created at init by a FEM for example, and then they are managed by the TopologyContainer. In a classic simulation, at the end, Topology is deleted and the set of topologyHandler is only clear (not deleted), and when the FEM is deleted, the topologyHandler is deleted as well.

But in the scenario we checked where the mouse pickHandler is crashing, in fact a springFF is creating, adding a new TopologyHandler and when you release the mouse interaction, the springFF is deleted, which also delete the topologyHandler but not removed from TopologyContainer list. So in a incoming PR (to not mix all the changed, I thought this one would be quickly merged) I'm adding all the mechanism to unregister a TopologyHandler when it is destroyed.

But this raised another problem when the scene is deleted. To unregister itself, the handler need to access the topologyContainer which could have already been destroyed (depending on the graph order). That's why if the TopologyContainer is deleted first it sets all its topologyHander as unregistered.  

Using SPtr would have work as expected I think but now I don't remember why I couldn't use it (probably because we don't know which class of topologyContainer will be used/find in the scene)

Reply:
Full picture available in #3369